net/net.go

```diff
@@ -954,22 +1040,27 @@ func (n *net) loadRecords(
 	tid thread.ID,
 	lid peer.ID,
 	recs []core.Record,
-) ([]core.ThreadRecord, cid.Cid, error) {
+	counter int64,
+) ([]core.ThreadRecord, thread.Head, error) {
 	if len(recs) == 0 {
-		return nil, cid.Undef, errors.New("cannot load empty record chain")
+		return nil, thread.HeadUndef, errors.New("cannot load empty record chain")
+	}
+	head, err := n.currentHead(tid, lid)
+	if err != nil {
+		return nil, thread.HeadUndef, err
 	}
 
 	// check if the last record was already loaded and processed
 	var last = recs[len(recs)-1]
-	if exist, err := n.isKnown(last.Cid()); err != nil {
-		return nil, cid.Undef, err
-	} else if exist || !last.Cid().Defined() {
-		return nil, cid.Undef, nil
-	}
-
-	head, err := n.currentHead(tid, lid)
-	if err != nil {
-		return nil, head, err
+	// if we don't have the counter (but have some recs) then we were communicating with old version peer
```

Comment:
As far as I understand in situation with `AddRecord` we should basically use old checks for `isKnown`, because it is not possible for us to know the counter of record. This is not ideal of course, but we will not be able to fix it, unless the counter would be a part of record.